### PR TITLE
Use cookie to share subscription access token on DDG domains

### PR DESCRIPTION
**Pull request by miasma13**
Task/Issue URL:  https://app.asana.com/0/1108686900785972/1208264562025859/f

**Description**:
Store and keep in sync the subscription access token for the duckduckgo.com domain cookie. For the implementation guidelines please see description of the linked task.

**Steps to test this PR**:
To inspect website cookies use "Cookies" in the debug menu. Please mind that cookies may appear empty with no WebView being instantiated (no tabs), ensure there is at least tab with webpage open (unless required not to).
To test purchase use the real device and test the Alpha target.

1. Ensure no subscription is on device
2. Open SERP and then check cookies via debug menu -> subscription cookie should not be present
3. Purchase or restore a subscription
4. Open SERP and then check cookies via debug menu -> subscription cookie should be present
5. Restart the browser
6. Open SERP and then check cookies via debug menu -> subscription cookie should still be present
7. Remove subscription
8. Open SERP and then check cookies via debug menu -> subscription cookie should not be present
9. Restart the browser
10. Open SERP and then check cookies via debug menu -> subscription cookie should still not be present

Test for proper handling of subscription cookie while no WebKit stack is in place (no tabs)

1. Ensure no subscription is on device
2. Ensure auto clearing is enable
3. Restart the browser
4. Do not open any tabs
5. Purchase or restore a subscription
6. Open SERP and then check cookies via debug menu -> subscription cookie should be present
7. Restart the browser
8. Do not open any tabs
9. Remove subscription
10. Open SERP and then check cookies via debug menu -> subscription cookie should not be present

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/AppDelegate.swift [hunk 1]
```diff
@@ -88,6 +88,7 @@ import os.log
     private var autofillUsageMonitor = AutofillUsageMonitor()
 
     private(set) var subscriptionFeatureAvailability: SubscriptionFeatureAvailability!
+    private(set) var subscriptionCookieManager: SubscriptionCookieManaging!
```
Comment (ankushkushwaha) [inline DuckDuckGo/AppDelegate.swift]: Should we avoid force unwraping, it might cause the app crash ?

File: DuckDuckGo/TabViewController.swift [hunk 2]
```diff
@@ -636,6 +636,8 @@ class TabViewController: UIViewController {
             await webView.configuration.websiteDataStore.dataRecords(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes())
             let cookieStore = webView.configuration.websiteDataStore.httpCookieStore
             await WebCacheManager.shared.consumeCookies(httpCookieStore: cookieStore)
+            (UIApplication.shared.delegate as? AppDelegate)?.subscriptionCookieManager.resetLastRefreshDate()
+            await (UIApplication.shared.delegate as? AppDelegate)?.subscriptionCookieManager.refreshSubscriptionCookie()
```
Comment (brindy) [inline DuckDuckGo/TabViewController.swift]: Why can't subscription manager be injected?

**Review by brindy — CHANGES_REQUESTED** [review]
Looks good, but please inject the subscription manager rather than use the singleton.  Let's not continue to compound that tech debt :)

**Review by brindy — APPROVED** [review]
LGTM

**Pull request merged**